src/Collections.ts

```diff
@@ -209,6 +209,11 @@ export abstract class EnumerableCollection<TElement>
         return this.asEnumerable().singleOrDefault();
     }
 
+    public skipWhile(predicate: Predicate<TElement>): IEnumerable<TElement>
```

Comment:
add loop based implementation in `ArrayEnumerable` (I just noticed that implementation for `where` is also missing (`filter` will do the trick here))

Reply:
Hmm, i'm not sure what i need to do here. Do i add `skipWhile` method to `ArrayEnumerable`? If that's the case why can't i do it like this?

    public skipWhile(predicate: Predicate<TOut>): IEnumerable<TOut>
    {
        return this.list.skipWhile(predicate);
    }

Reply:
oh sorry, my fault, his doesn't affect deferred-executed methods. The idea behind `ArrayEnumerable` is to create optimized versions of the non-deferred algorithms. Not this case.